doc/design/block.md

```diff
@@ -0,0 +1,210 @@
+# Design Doc: Use Block in RNNOp, While Op, IfElseOp
+
+In C++ and Java programming language, a block is a lexical structure of source code which is grouped as one line of code.
+
+RNNOp looks like the loop structure in programming languages.
+And similarly, WhileOp and IfElseOp are like loop and conditions respectively.
+So we want to verify if we should have a class Block in PaddlePaddle that works like a pair of curly braces in the loop and condition structures of programming languages.
+
+Blocks do not only group source code, but also narrow the lexical scope of variables so that they do not conflict with variables having the same name used elsewhere in a program.
+
+In Paddle, we need a similar concept called Block to support following scenes:
+
+- define a PaddlePaddle program by writing blocks of codes, which includes the definitions of variables and operators.
+  - `RNNOp`, `SwitchOp`, `WhileOp` and `IfElseOp`, etc, need Block to help to define sub-block. 
+- help to execute multiple operators, blocks should group operators and runs like a single operator.
+
```

Comment:
This statement is not reasonable.
According to the design, 
> a block is a lexical structure of source code which is grouped as one line of code

We already differentiate the static phase and dynamic phase.  In the words declared above, `Block` is designed for static phase. So it has nothing to do with dynamic phase.  which means, block only `group operators`, but does not know their execute order, be partitioned into two `Block` or not.